src/routes/v2/slp.ts

```diff
@@ -1900,17 +1960,165 @@ async function txDetails(
       return res.json({ error: msg })
     }
 
-    // Handle corner case of mis-typted txid
-    if (err.error.indexOf("Not found") > -1) {
-      res.status(400)
-      return res.json({ error: "TXID not found" })
-    }
+    // // Handle corner case of mis-typted txid
+    // if (err.error.indexOf("Not found") > -1) {
+    //   res.status(400)
+    //   return res.json({ error: "TXID not found" })
+    // }
 
     res.status(500)
     return res.json({ error: util.inspect(err) })
   }
 }
 
+// Manipulates and formats the raw data comming from Insight API.
+const processInputs = (tx: TransactionInterface): any => {
+  // Add legacy and cashaddr to tx vin
+  if (tx.vin) {
+    tx.vin.forEach((vin: any): any => {
+      if (!vin.coinbase) {
+        vin.value = vin.valueSat
+        const address: string = vin.addr
+        if (address) {
+          vin.legacyAddress = bitbox.Address.toLegacyAddress(address)
+          vin.cashAddress = bitbox.Address.toCashAddress(address)
+          delete vin.addr
+        }
+        delete vin.valueSat
+        delete vin.doubleSpentTxID
+      }
+    })
+  }
+
+  // Add legacy and cashaddr to tx vout
+  if (tx.vout) {
+    tx.vout.forEach((vout: any): any => {
+      // Overwrite value string with value in satoshis
+      //vout.value = parseFloat(vout.value) * 100000000
+
+      if (vout.scriptPubKey) {
+        if (vout.scriptPubKey.addresses) {
+          const cashAddrs: string[] = []
+          vout.scriptPubKey.addresses.forEach((addr: any) => {
+            const cashAddr = bitbox.Address.toCashAddress(addr)
+            cashAddrs.push(cashAddr)
+          })
+          vout.scriptPubKey.cashAddrs = cashAddrs
+        }
+      }
+    })
+  }
+}
+
+// Retrieve transaction data from the Insight API
+// This function is also used by the SLP route library.
+async function transactionsFromInsight(txid: string): Promise<any> {
```

Comment:
Rather than duplicate code, I think the best practice would be to `import` the `transactions.ts` class and call the function from there.